File: matiasmasso/Mat.NET/DT2/DTORol.vb

Public Class DTORol

    Property id As Ids

    Property nom As DTOLangText

    Public Property dsc As String

    Property isLoaded As Boolean
    Property isNew As Boolean

    Public Enum Ids
        NotSet
        SuperUser
        Admin
        Operadora
        Accounts
        SalesManager
        Comercial
        Taller
        Rep = 8
        CliFull = 9
        Manufacturer = 10
        Marketing = 11
        CliLite = 12
        Auditor = 14
        Banc = 15
        Transportista = 16
        LogisticManager = 17
        PR = 18
        Guest = 20
        Lead = 98
        Unregistered = 99
        Denied = 100
    End Enum

    Public Sub New()
        MyBase.New()
        _Id = Ids.Unregistered
        _Nom = New DTOLangText
    End Sub

    Public Sub New(oId As Ids)
        MyBase.New()
        _Id = oId
        _Nom = New DTOLangText
    End Sub

    Public ReadOnly Property NomEsp As String
        Get
            Return _Nom.Esp
        End Get
    End Property

    Public Function IsAuthenticated() As Boolean
        Dim retval As Boolean
        Select Case _Id
            Case DTORol.Ids.NotSet, DTORol.Ids.Unregistered, DTORol.Ids.Denied
            Case Else
                retval = True
        End Select
        Return retval
    End Function


    Public Function IsSuperAdmin() As Boolean
        Dim Retval As Boolean = False
        Select Case _Id
            Case Ids.SuperUser
                Retval = True
        End Select
        Return Retval
    End Function

    Public Function IsAdmin() As Boolean
        Dim Retval As Boolean = False
        Select Case _Id
            Case Ids.SuperUser, Ids.Admin
                Retval = True
        End Select
        Return Retval
    End Function

    Public Function IsMainboard() As Boolean
        Dim Retval As Boolean = False
        Select Case _Id
            Case Ids.SuperUser, Ids.Admin, Ids.SalesManager
                Retval = True
        End Select
        Return Retval
    End Function

    Public Function IsStaff(Optional ByVal oRolId As Ids = Ids.NotSet) As Boolean
        If oRolId = Ids.NotSet Then oRolId = _Id

        Dim oRetVal As Boolean = False
        Select Case oRolId
            Case Ids.SuperUser,
            Ids.Admin,
            Ids.SalesManager,
             Ids.Accounts,
              Ids.LogisticManager,
               Ids.Marketing,
               Ids.Operadora,
                Ids.Comercial,
                 Ids.Taller

                oRetVal = True
        End Select
        Return oRetVal
    End Function

    Public Function IsRep(Optional ByVal oRolId As Ids = Ids.NotSet) As Boolean
        If oRolId = Ids.NotSet Then oRolId = _Id

        Dim oRetVal As Boolean = False
        Select Case oRolId
            Case Ids.Rep,
            Ids.Comercial

                oRetVal = True
        End Select

        Return oRetVal
    End Function

    Public Function IsProfesional() As Boolean
        Dim retval As Boolean
        Select Case _Id
            Case Ids.Denied, Ids.Guest, Ids.Lead, Ids.NotSet, Ids.Unregistered
            Case Else
                retval = True
        End Select
        Return retval
    End Function

    Public Shadows Function Equals(oCandidate As Object) As Boolean
        Dim retval As Boolean
        If oCandidate IsNot Nothing Then
            If TypeOf oCandidate Is DTORol Then
                retval = (oCandidate.id = _Id)
            End If
        End If
        Return retval
    End Function
End Class

File: matiasmasso/Mat.NET/DTO/DTORol.cs
namespace DTO
{
    public class DTORol
    {
        public Ids id { get; set; } //lowercase due to dependency on IOS iMat 24/3/2022

        public DTOLangText Nom { get; set; }

        public string Dsc { get; set; }

        public bool IsLoaded { get; set; }
        public bool IsNew { get; set; }

        public enum Ids
        {
            notSet,
            superUser,
            admin,
            operadora,
            accounts,
            salesManager,
            comercial,
            taller,
            rep = 8,
            cliFull = 9,
            manufacturer = 10,
            marketing = 11,
            cliLite = 12,
            auditor = 14,
            banc = 15,
            transportista = 16,
            logisticManager = 17,
            pr = 18,
            guest = 20,
            employee = 21,
            lead = 98,
            unregistered = 99,
            denied = 100
        }

        public DTORol() : base()
        {
            id = Ids.unregistered;
            Nom = new DTOLangText();
        }

        public DTORol(Ids oId) : base()
        {
            id = oId;
            Nom = new DTOLangText();
        }

        public string nomEsp
        {
            get
            {
                return Nom.Esp;
            }
        }

        public bool isAuthenticated()
        {
            bool retval = false;
            switch (id)
            {
                case DTORol.Ids.notSet:
                case DTORol.Ids.unregistered:
                case DTORol.Ids.denied:
                    {
                        break;
                    }

                default:
                    {
                        retval = true;
                        break;
                    }
            }
            return retval;
        }


        public bool isSuperAdmin()
        {
            bool Retval = false;
            switch (id)
            {
                case Ids.superUser:
                    {
                        Retval = true;
                        break;
                    }
            }
            return Retval;
        }

        public bool isAdmin()
        {
            bool Retval = false;
            switch (id)
            {
                case Ids.superUser:
                case Ids.admin:
                    {
                        Retval = true;
                        break;
                    }
            }
            return Retval;
        }

        public bool isMainboard()
        {
            bool Retval = false;
            switch (id)
            {
                case Ids.superUser:
                case Ids.admin:
                case Ids.salesManager:
                    {
                        Retval = true;
                        break;
                    }
            }
            return Retval;
        }

        public bool isStaff(Ids oRolId = Ids.notSet)
        {
            if (oRolId == Ids.notSet)
                oRolId = id;

            bool oRetVal = false;
            switch (oRolId)
            {
                case Ids.superUser:
                case Ids.admin:
                case Ids.salesManager:
                case Ids.accounts:
                case Ids.logisticManager:
                case Ids.marketing:
                case Ids.operadora:
                case Ids.comercial:
                case Ids.taller:
                    {
                        oRetVal = true;
                        break;
                    }
            }
            return oRetVal;
        }

        public bool isRep(Ids oRolId = Ids.notSet)
        {
            if (oRolId == Ids.notSet)
                oRolId = id;

            bool oRetVal = false;
            switch (oRolId)
            {
                case Ids.rep:
                case Ids.comercial:
                    {
                        oRetVal = true;
                        break;
                    }
            }

            return oRetVal;
        }

        public bool isCustomer(Ids oRolId = Ids.notSet)
        {
            if (oRolId == Ids.notSet)
                oRolId = id;

            bool oRetVal = false;
            switch (oRolId)
            {
                case Ids.cliFull:
                case Ids.cliLite:
                    {
                        oRetVal = true;
                        break;
                    }
            }

            return oRetVal;
        }

        public bool isProfesional()
        {
            bool retval = false;
            switch (id)
            {
                case Ids.denied:
                case Ids.guest:
                case Ids.lead:
                case Ids.notSet:
                case Ids.unregistered:
                    {
                        break;
                    }

                default:
                    {
                        retval = true;
                        break;
                    }
            }
            return retval;
        }

        public new bool Equals(object oCandidate)
        {
            bool retval = false;
            if (oCandidate != null)
            {

                if (oCandidate.GetType() == typeof(DTORol))
                {
                    DTORol oCandidateRol = (DTORol)oCandidate;
                    retval = (oCandidateRol.id == this.id);
                }

            }
            return retval;
        }
    }
}
